processing/src/test/java/org/apache/druid/concurrent/ExecsTest.java

```diff
@@ -118,6 +123,37 @@ public void run()
     producer.shutdown();
   }
 
+  @Test
+  public void testSynchronousQueueSingleThreadedExecutor() throws Exception
```

Comment:
What does this test verify?

Reply:
This test verifies, that if after `shutDownNow()` is called on executor service, a task is submitted. It will not stop the executor service from shutting down. If we remove the fix in Execs, this will get STUCK in waiting state. 
I can improve this test if wanted

Reply:
No, the test should be fine. Maybe try to update the test name to reflect what it verifies.